manifests/haproxy_exporter.pp

```diff
@@ -73,15 +73,15 @@
 #  The binary release version
 
 class prometheus::haproxy_exporter(
-  Variant[Stdlib::HTTPUrl, Stdlib::HTTPSUrl] $cnf_scrape_uri,
+  Variant[Stdlib::HTTPUrl, Pattern[/unix:(?:\/.+)+/]] $cnf_scrape_uri,
```

Comment:
* Why did you remove the https type?
* Would https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/types/unixpath.pp work as well for a unix socket?

Reply:
I removed https type because Stdlib::HTTPUrl covers both http and httpS, see https://github.com/puppetlabs/puppetlabs-stdlib

Reply:
Unixpath doesn't work for path as it doesn't allow ":" in the pattern.
Example of what I'm currently using: `unix:/var/lib/haproxy/stats`